javascript/javascript3/week3/fetchgithubapirepo.js

```diff
@@ -0,0 +1,78 @@
+/*const url = 'https://api.github.com/search/repositories?q=user:benna100'
+
+fetch(url)
+  .then(res => res.json())
+  .then(result => {
+    console.log(result) //to check the data on server
+    const gitData = result.items
+    let gitFullName = gitData.map((m) => {
+      return m.full_name
+    })
+    let gitHtmlUrl = gitData.map((m) => {
+      return m.html_url
+    })
+    let gitOwner = gitData.map((m) => {
+      return m.owner
+    })
+  })
+  */
+/*
+const student1 = 'https://api.github.com/search/repositories?q=user:benna100';
+const student2 = 'https://api.github.com/search/repositories?q=user:panivita';
+const student3 = 'https://api.github.com/search/repositories?q=user:LucyChyzhova';
+
+const student1url = fetch(student1);
+const student2url = fetch(student2);
+const student3url = fetch(student3);
+
+const allStudentsUrl = [student1url, student2url, student3url]
+Promise.all(allStudentsUrl)
+  .then(res => {
+    console.log(res)
+
+    const jsonResult1 = res[0].json()
+    const jsonResult2 = res[1].json()
+    const jsonResult3 = res[2].json()
+    return Promise.all([jsonResult1, jsonResult2, jsonResult3])
+  })
+  .then(result => {
+    console.log(result)
+    const gitData = result.items
+    let gitFullName = gitData.map((m) => {
+      return m.full_name
+    })
+    console.log(gitFullName[0])
+    let gitHtmlUrl = gitData.map((m) => {
+      return m.html_url
+    })
+    let gitOwner = gitData.map((m) => {
+      return m.owner
+    })
+  })
+*/
+
+const student1 = "https://api.github.com/search/repositories?q=user:benna100";
+const student2 = "https://api.github.com/search/repositories?q=user:panivita";
+const student3 =
+  "https://api.github.com/search/repositories?q=user:LucyChyzhova";
```

Comment:
You could do `const url = "https://api.github.com/search/repositories?q=user:"`, and then add names to `url` to make the code reusable. It would be useful in case there are many students, not just 3.